--- a/inst/include/xla/pjrt/c/pjrt_c_api.h
+++ b/inst/include/xla/pjrt/c/pjrt_c_api.h
@@ -3024,7 +3024,7 @@
 
 // -------------------------------- API access ---------------------------------
 
-#define _PJRT_API_STRUCT_FIELD(fn_type) fn_type* fn_type
+#define _PJRT_API_STRUCT_FIELD(fn_type) fn_type* fn_type##_
 
 // Please modify PJRT_Api_STRUCT_SIZE if the last field of PJRT_Api is changed.
 typedef struct PJRT_Api {
@@ -3223,7 +3223,7 @@
 
 enum {
   PJRT_Api_STRUCT_SIZE =
-      PJRT_STRUCT_SIZE(PJRT_Api, PJRT_TopologyDescription_GetMemorySpaceKindIds)
+      PJRT_STRUCT_SIZE(PJRT_Api, PJRT_TopologyDescription_GetMemorySpaceKindIds_)
 };
 
 #undef _PJRT_API_STRUCT_FIELD
