diff --git a/inst/include/xla/ffi/api/api.h b/inst/include/xla/ffi/api/api.h
index 20e927d..4885c3b 100644
--- a/inst/include/xla/ffi/api/api.h
+++ b/inst/include/xla/ffi/api/api.h
@@ -190,6 +190,7 @@ inline std::ostream& operator<<(std::ostream& os,
     case XLA_FFI_ExecutionStage_EXECUTE:
       return os << "execute";
   }
+  throw std::runtime_error("unknown execution stage");
 }
 
 namespace xla::ffi {
